core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/buffer_v1.go

```diff
@@ -124,6 +129,17 @@ func (b *logBuffer) Enqueue(uid *big.Int, logs ...logpoller.Log) (int, int) {
 	return buf.enqueue(blockThreshold, logs...)
 }
 
+func (b *logBuffer) evictReorgdLogs(reorgBlocks map[int64]bool) {
+	for _, queue := range b.queues {
+		for blockNumber := range reorgBlocks {
+			if _, ok := queue.logs[blockNumber]; ok {
+				queue.logs[blockNumber] = []logpoller.Log{}
+				b.dequeueCoordinator.markReorg(blockNumber, b.opts.blockRate.Load())
```

Comment:
this call is a bit redundant as it is made multiple times for all upkeeps in buffer

Reply:
Rearranged the loops so that we only call this once per block number